File: backend/components/metiers/threed/tests_centralises/fixtures/mocks/test_fixtures_mock.py
"""
test___init__.py – Tests ultra avancés pour les mocks fixtures (Python)
"""

from backend.components.metiers.threed.fixtures.mocks.fixtures_mock import (
    mock_request,
    mock_response,
)


def test_import_fixtures_mock():
    assert callable(mock_request)
    assert callable(mock_response)


def test_mock_request(mock_request):
    req = mock_request
    assert req["headers"]["x-test"] == "true"
    assert req["user"]["id"] == "mock-user"
    assert req["user"]["role"] == "test"


def test_mock_response(mock_response):
    resp = mock_response
    assert resp.status_code == 200

File: backend/components/metiers/threed/tests_centralises/fixtures/mocks/test_fixtures.mock.js
// test_fixtures.mock.js – Tests ultra avancés pour les mocks fixtures (JS)
const { mockRequest, mockResponse } = require('./fixtures.mock');

test('mockRequest retourne une requête mockée', () => {
  const req = mockRequest({ foo: 'bar' });
  expect(req.headers['x-test']).toBe('true');
  expect(req.user.id).toBe('mock-user');
  expect(req.user.role).toBe('test');
  expect(req.body.foo).toBe('bar');
});

test('mockResponse retourne une réponse mockée', () => {
  const res = mockResponse();
  expect(typeof res.status).toBe('function');
  expect(typeof res.json).toBe('function');
  expect(res.status()).toBe(res);
  expect(res.json()).toBe(res);
});
